ethdb/privateapi/ethbackend.go

```diff
@@ -271,18 +282,31 @@ func (s *EthBackendServer) EngineExecutePayloadV1(ctx context.Context, req *type
 
 // EngineGetPayloadV1, retrieves previously assembled payload (Validators only)
 func (s *EthBackendServer) EngineGetPayloadV1(ctx context.Context, req *remote.EngineGetPayloadRequest) (*types2.ExecutionPayload, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	// Wait some time in case validition process has not started
```

Comment:
Why wait here?

Reply:
because payload assembly is asynchronous, which means that if the CL requests the payload immediately after EngineForkChoiceUpdatedV1 we may return unknown payload, when we indeed did not have enough time to process it.

Reply:
this is the case with mergemock for example